src/core_ocean/shared/mpas_ocn_diagnostics.F

```diff
@@ -406,6 +407,7 @@ subroutine ocn_diagnostic_solve(dt, statePool, forcingPool, meshPool, diagnostic
            end do
          end do
          !$omp end do
+         call mpas_threading_barrier()
```

Comment:
This barrier is not necessary, because there is an implied synchronization among threads at the end of the loop. When needed, such barriers can be removed with
```
!$omp end do nowait
```
and threads do not wait after exiting the loop.

Reply:
Thanks for watching this repo, @amametjanov! That is really helpful to know.  I added that thread barrier because I thought it was needed, not because I saw a difference in a test.

Reply:
Update: [this PR](https://github.com/amametjanov/MPAS-Model/pull/3) (a follow-up to PR #152) should reduce the number thread barriers in the remaining ocean sources and improve threading throughput. Thanks.